src/Geno/C++/GUI/Widgets/TitleBar.cpp

```diff
@@ -213,8 +215,88 @@ void TitleBar::Draw( void )
 				ButtonRect.Max.x -= ButtonSize;
 			}
 		}
-#endif
+	#elif defined ( __linux__ )
+		{
+			GLFWwindow* pWindow          = MainWindow::Instance().GetWindow();
+			Display* pX11Display         = glfwGetX11Display();
+			Window      X11Window        = glfwGetX11Window( pWindow );
+			ImVec2      CursorPos        = ImGui::GetMousePos();
+			int         WindowSize[ 2 ]  ={ 0, 0 };
+			int         Border           = 5;
+
+			glfwGetWindowSize( pWindow, &WindowSize[ 0 ], &WindowSize[ 1 ] );
+
+			int rx, ry;
+
+			XWindowAttributes win_attributes;
+
+			Window junkwin;
+
+			XGetWindowAttributes( pX11Display, X11Window, &win_attributes );
 
+			// Get Top left
+			XTranslateCoordinates( pX11Display, X11Window, win_attributes.root, -win_attributes.border_width, -win_attributes.border_width, &rx, &ry, &junkwin );
+
+			ImRect WindowImRect;
+			WindowImRect.Min ={ static_cast< float >( rx ), static_cast< float >( ry ) };
+			WindowImRect.Max ={ static_cast< float >( WindowSize[ 0 ] ), static_cast< float >( WindowSize[ 1 ] ) };
+
+
+			if( ImGui::IsMousePosValid( &CursorPos ) )
+			{
+				// Top
+				if( CursorPos.y < ( WindowImRect.GetTR().y + Border ) )
+				{
+					if( CursorPos.x >= ( WindowImRect.GetTR().x - Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNESW );
+						ResizeWindow( pWindow, 6 );
+					}
+				}
+				if( CursorPos.y < ( WindowImRect.GetTL().y + Border ) )
+				{
+					if( CursorPos.x < ( WindowImRect.GetTL().x + Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNWSE );
+						ResizeWindow( pWindow, 5 );
+					}
+				}
+				// Bottom
+				if( CursorPos.y >= ( WindowImRect.GetBR().y + Border ) )
+				{
+					if( CursorPos.x >= ( WindowImRect.GetBR().x - Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNWSE );
+						ResizeWindow( pWindow, 8 );
+					}
+				}
+				if( CursorPos.y >= ( WindowImRect.GetBL().y + Border ) )
+				{
+					if( CursorPos.x < ( WindowImRect.GetBL().x + Border ) )
+					{
+						ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeNESW );
+						ResizeWindow( pWindow, 7 );
+					}
+				}
+				else if( CursorPos.x < ( WindowImRect.Min.x + Border ) )
+				{
+					ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeEW );
+					ResizeWindow( pWindow, 1 );
+				}
+				else if( CursorPos.x >= ( WindowImRect.Max.x - Border ) )
+				{
+					ImGui::SetMouseCursor( ImGuiMouseCursor_ResizeEW );
+					ResizeWindow( pWindow, 3 );
+				}
+				else
+				{
+					// Drag the menu bar to move the window
+					if( !ImGui::IsAnyItemHovered() && ( CursorPos.y < ( WindowImRect.Min.y + m_Height ) ) && ImGui::IsMouseClicked( ImGuiMouseButton_Left ) )
+						DragWindow( pWindow );
+				}
+			}
```

Comment:
This can be refactored to reduce the number of if-statements.
For example:
`if( CursorPos.y < ( WindowImRect.GetTR().y + Border ) )` is the same thing as
`if( CursorPos.y < ( WindowImRect.GetTL().y + Border ) )` because both TR.y and TL.y is the Y level of the Top.